network/datadog_checks/network/network.py

```diff
@@ -290,32 +290,32 @@ def _check_linux(self, instance):
             try:
                 self.log.debug("Using `ss` to collect connection state")
                 # Try using `ss` for increased performance over `netstat`
+                metrics = self._get_metrics()
                 for ip_version in ['4', '6']:
-                    for protocol in ['tcp', 'udp']:
-                        # Call `ss` for each IP version because there's no built-in way of distinguishing
-                        # between the IP versions in the output
-                        # Also calls `ss` for each protocol, because on some systems (e.g. Ubuntu 14.04), there is a
-                        # bug that print `tcp` even if it's `udp`
-                        output, _, _ = get_subprocess_output(
-                            ["ss", "-n", "-{0}".format(protocol[0]), "-a", "-{0}".format(ip_version)], self.log
-                        )
-                        lines = output.splitlines()
-
-                        # State      Recv-Q Send-Q     Local Address:Port       Peer Address:Port
-                        # UNCONN     0      0              127.0.0.1:8125                  *:*
-                        # ESTAB      0      0              127.0.0.1:37036         127.0.0.1:8125
-                        # UNCONN     0      0        fe80::a00:27ff:fe1c:3c4:123          :::*
-                        # TIME-WAIT  0      0          90.56.111.177:56867        46.105.75.4:143
-                        # LISTEN     0      0       ::ffff:127.0.0.1:33217  ::ffff:127.0.0.1:7199
-                        # ESTAB      0      0       ::ffff:127.0.0.1:58975  ::ffff:127.0.0.1:2181
-
-                        metrics = self._parse_linux_cx_state(
-                            lines[1:], self.tcp_states['ss'], 0, protocol=protocol, ip_version=ip_version
-                        )
-                        # Only send the metrics which match the loop iteration's ip version
-                        for stat, metric in iteritems(self.cx_state_gauge):
-                            if stat[0].endswith(ip_version) and stat[0].startswith(protocol):
-                                self.gauge(metric, metrics.get(metric), tags=custom_tags)
+                    # Call `ss` for each IP version because there's no built-in way of distinguishing
+                    # between the IP versions in the output
+                    # Also calls `ss` for each protocol, because on some systems (e.g. Ubuntu 14.04), there is a
```

Comment:
Is it still doing this?

Reply:
Yes? We kept the 4 calls to `ss`, just not in a for loop for protocols.